test/test_io.py

```diff
@@ -2681,6 +2682,44 @@ def test_mean_axis(self):
         self.assertIsInstance(m, FlowCal.io.FCSData)
         self.assertEqual(m.shape, m_array.shape)
         np.testing.assert_array_equal(m, m_array)
-        
+
+class TestFCSDataPickle(unittest.TestCase):
+    '''See Issue #277. FCSData objects must be able to be serialized (via Pickle)
+    and deserialized correctly.'''
+
+    def setUp(self):
+        self.d = FlowCal.io.FCSData(filenames[2])
+        file_dir = os.path.dirname(os.path.abspath(__file__))
+        self.test_file = os.path.join(file_dir, 'test_fcs.p')
+
+    def test_pickle_unpickle(self):
+        """
+        Test that an FCSData instance can be pickled, then unpickled, and
+        still retain all custom attributes.
+        """
+        channels = self.d.channels # Should be fine, no error
+
+        with open(self.test_file, 'w') as outfile:
+            cPickle.dump(self.d, outfile) # serialization
+
+        with open(self.test_file, 'r') as infile:
+            test_fcs = cPickle.load(infile) # de-serialization
+
+        # Check all attributes/methods are preserved
```

Comment:
This section is checking that the same attributes exist, and that some subset of the hidden attributes have the same value. I would recommend that the contents of the array and the actual values of computed attributes are also checked. Even if you know that these worked before this PR, it would be a good idea for the tests to be as comprehensive as possible in order to detect future code changes that break proper pickling.

For the array values, you might want to use [`assert_array_equal`](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.testing.assert_array_equal.html) or [`assert_almost_equal`](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.testing.assert_almost_equal.html).

Reply:
These are good ideas - thanks! I believe I've incorporated your suggestions, but please let me know if there are still cases/attributes/data that I'm forgetting!